src/functional/iterator/transforms/propagate_deref.py

```diff
@@ -0,0 +1,45 @@
+from eve import NodeTranslator
+from eve.pattern_matching import ObjectPattern as P
+from functional.iterator import ir
+
+
+# TODO(tehrengruber): This pass can be generalized to all builtins, e.g.
+#  `plus((λ(...) → plus(...))(...), ...)` can be transformed into
```

Comment:
isn't the inner plus confusing, i.e. unnecessary for what you describe?

Reply:
True. Fixed.